rust/src/main.rs

```diff
@@ -0,0 +1,117 @@
+            }
+        }
+        if language.is_none() {
+            println!("Please specify a language!");
+            usage(&args[0]);
+            return;
+        }
+        let stemmer = Stemmer::create(language.unwrap());
+        
+
+        let mut output = if let Some(output_file) = output_arg {
+            Box::new(File::create(Path::new(&output_file)).unwrap()) as Box<Write>
+        } else {
+            Box::new(std::io::stdout()) as Box<Write>
+        };
+
+        if let Some(input_file) = input_arg {
+            for line in BufReader::new(File::open(Path::new(&input_file)).unwrap()).lines() {
+                writeln!(&mut output, "{}", stemmer.stem(&line.unwrap())).unwrap();
+            }
+        } else {
+            let stdin = std::io::stdin();
+            for line in stdin.lock().lines() {
+                writeln!(&mut output, "{}", stemmer.stem(&line.unwrap())).unwrap();
+            }
+        }        
+    }
+}
+
+
+/// Wrapps a usable interface around the actual stemmer implementation
+pub struct Stemmer {
+    stemmer: Box<Fn(&mut SnowballEnv) -> bool>,
+}
+
+impl Stemmer {
+    /// Create a new stemmer from an algorithm
+    pub fn create(lang: String) -> Self {
+        use snowball::algorithms;
+        match lang.as_str() {
+            "arabic" => Stemmer { stemmer: Box::new(algorithms::arabic_stemmer::stem) },
+            "danish" => Stemmer { stemmer: Box::new(algorithms::danish_stemmer::stem) },
+            "dutch" => Stemmer { stemmer: Box::new(algorithms::dutch_stemmer::stem) },
+            "english" => Stemmer { stemmer: Box::new(algorithms::english_stemmer::stem) },
+            "finnish" => Stemmer { stemmer: Box::new(algorithms::finnish_stemmer::stem) },
+            "french" => Stemmer { stemmer: Box::new(algorithms::french_stemmer::stem) },
+            "german" => Stemmer { stemmer: Box::new(algorithms::german_stemmer::stem) },
+            "hungarian" => Stemmer { stemmer: Box::new(algorithms::hungarian_stemmer::stem) },
+            "italian" => Stemmer { stemmer: Box::new(algorithms::italian_stemmer::stem) },
+            "norwegian" => Stemmer { stemmer: Box::new(algorithms::norwegian_stemmer::stem) },
+            "porter" => Stemmer { stemmer: Box::new(algorithms::porter_stemmer::stem) },
+            "portuguese" => Stemmer { stemmer: Box::new(algorithms::portuguese_stemmer::stem) },
+            "romanian" => Stemmer { stemmer: Box::new(algorithms::romanian_stemmer::stem) },
+            "russian" => Stemmer { stemmer: Box::new(algorithms::russian_stemmer::stem) },
+            "spanish" => Stemmer { stemmer: Box::new(algorithms::spanish_stemmer::stem) },
+            "swedish" => Stemmer { stemmer: Box::new(algorithms::swedish_stemmer::stem) },
+            "tamil" => Stemmer { stemmer: Box::new(algorithms::tamil_stemmer::stem) },
+            "turkish" => Stemmer { stemmer: Box::new(algorithms::turkish_stemmer::stem) },
+            x => panic!("Unkown algorithm '{}'", x)
```

Comment:
Instead of hard-coding this mapping, generate it automatically. See for example how python/create_init.py creates the equivalent mapping for Python.

Also, “Unkown” should be “Unknown”.